contracts/multisig-prover/src/encoding/bcs.rs

```diff
@@ -0,0 +1,284 @@
+                            HexBinary::from_hex("023e44013597b8a49df193265ae443e1a9970626d4df92e4ebad677ab2aca5c13a").unwrap(),
+                        )
+                            .try_into()
+                            .unwrap(),
+                        weight: Uint128::one(),
+                    },
+                ),
+                (
+                    "addr_3".to_string(),
+                    Signer {
+                        address: Addr::unchecked("addr_3"),
+                        pub_key: (
+                            KeyType::Ecdsa,
+                            HexBinary::from_hex("024fa6a34ec85dc2618d730ad68ab914ccc492e54b573172083c0fa44465f54dcc").unwrap(),
+                        )
+                            .try_into()
+                            .unwrap(),
+                        weight: Uint128::one(),
+                    },
+                ),
+            ]
+            .into_iter()
+            .collect(),
+            threshold: 2u128.into(),
+            created_at: 2024,
+        };
+        let payload = Payload::VerifierSet(VerifierSet {
+            signers: vec![
+                (
+                    "addr_1".to_string(),
+                    Signer {
+                        address: Addr::unchecked("addr_1"),
+                        pub_key: (
+                            KeyType::Ecdsa,
+                            HexBinary::from_hex("0309613c4ae8b9ac87bdb3c4ff240a7e5f905f59754f377ccf54fbd8ce0e8ba636").unwrap(),
+                        )
+                            .try_into()
+                            .unwrap(),
+                        weight: Uint128::one(),
+                    },
+                ),
+                (
+                    "addr_2".to_string(),
+                    Signer {
+                        address: Addr::unchecked("addr_2"),
+                        pub_key: (
+                            KeyType::Ecdsa,
+                            HexBinary::from_hex("032b14344bda89d1a0f1a976af94648f1b4a5df5397d008f3b2032267c11eda7c9").unwrap(),
+                        )
+                            .try_into()
+                            .unwrap(),
+                        weight: Uint128::one(),
+                    },
+                ),
+                (
+                    "addr_3".to_string(),
+                    Signer {
+                        address: Addr::unchecked("addr_3"),
+                        pub_key: (
+                            KeyType::Ecdsa,
+                            HexBinary::from_hex("031fb4e7844794a28bc49e8a702c984031d8627befea844932a02e5e918f59f610").unwrap(),
+                        )
+                            .try_into()
+                            .unwrap(),
+                        weight: Uint128::one(),
+                    },
+                ),
+            ]
+            .into_iter()
+            .collect(),
+            threshold: 2u128.into(),
+            created_at: 2025,
+        });
+
+        assert_eq!(
```

Comment:
use goldie::assert_json! for testing against golden files (you can refer the existing usage)

Reply:
done